File: python/토마토(7576) copy.py
from collections import deque

m, n = tuple(map(int, input().split()))
graph = [
    list(map(int, input().split()))
    for _ in range(n)
]

dx = [-1, 1, 0, 0]
dy = [0, 0, -1, 1]

queue = deque()
# 토마토가 익은 곳이면 탐색 시작
for i in range(n):
    for j in range(m):
        if graph[i][j] == 1:
            queue.append((i,j))


def bfs():
    while queue:
        x, y = queue.popleft()

        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]

            if (nx >=0 and nx < n and ny >=0 and ny < m and graph[nx][ny] == 0):
                graph[nx][ny] = graph[x][y] + 1
                queue.append((nx,ny))

bfs()

ans = 0
for g in graph:
    for elem in g:
        if elem == 0:
            print(-1)
            exit()
    ans = max(ans, max(g))

print(ans-1)

File: js/토마토(7576) copy.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [m, n] = input[0].split(' ').map(Number);
const graph = input.slice(1).map((line) => line.split(' ').map(Number));

const dx = [-1,1,0,0];
const dy = [0,0,-1,1];
let queue = [];

for (let i=0; i<n; i++) {
    for (let j=0; j<m; j++) {
        if (graph[i][j] === 1) {
            queue.push([i,j]);
        }
    }
}

let front = 0;

function bfs() {
    while (front < queue.length) {
        const [x, y] = queue[front];
        front++;

        for (let i=0; i<4; i++) {
            const nx = x + dx[i];
            const ny = y + dy[i];

            if (nx >=0 && nx < n && ny >=0 && ny < m && graph[nx][ny] === 0) {
                graph[nx][ny] = graph[x][y] + 1;
                queue.push([nx,ny]);
            }
        }
    }
}

bfs();

let ans = 0;

for (let g of graph) {
    for (let elem of g) {
        if (elem === 0) {
            console.log(-1);
            return;
        }
    }
    ans = Math.max(ans, Math.max(...g));
}

console.log(ans-1);